src/SOS/Strike/hostcoreclr.cpp

```diff
@@ -263,16 +269,124 @@ HRESULT GetCoreClrDirectory(std::string& coreClrDirectory)
+    directory.append("*");
+    std::string versionFound;
+
+    WIN32_FIND_DATAA data;
+    HANDLE findHandle = FindFirstFileA(directory.c_str(), &data);
+
+    if (findHandle != INVALID_HANDLE_VALUE) 
+    {
+        int highestRevision = 0;
+        do
+        {
+            if (data.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)
+            {
+                int major = 0;
+                int minor = 0;
+                int revision = 0;
+                if (sscanf_s(data.cFileName, "%d.%d.%d", &major, &minor, &revision) == 3)
+                {
+                    if (major == majorFilter && minor == minorFilter)
+                    {
+                        if (revision >= highestRevision)
+                        {
+                            highestRevision = revision;
+                            versionFound.assign(data.cFileName);
+                        }
+                    }
+                }
+            }
+        } 
+        while (0 != FindNextFileA(findHandle, &data));
+
+        FindClose(findHandle);
+    }
+
+    if (versionFound.length() > 0)
+    {
+        hostRuntimeDirectory.append(versionFound);
+        return true;
+    }
+
+    return false;
+}
+
+#ifdef FEATURE_PAL
+const char *g_linuxPaths[] = {
+//  "/rh-dotnet22/root/usr/bin/dotnet/shared/Microsoft.NETCore.App",
+    "/rh-dotnet21/root/usr/bin/dotnet/shared/Microsoft.NETCore.App",
+    "/rh-dotnet20/root/usr/bin/dotnet/shared/Microsoft.NETCore.App",
+    "/usr/share/dotnet/shared/Microsoft.NETCore.App",
+};
+#endif
+
+/**********************************************************************\
+ * Returns the path to the coreclr to use for hosting and it's
+ * directory. Attempts to use the best installed version of the 
+ * runtime, otherwise it defaults to the target's runtime version.
+\**********************************************************************/
 HRESULT GetHostRuntime(std::string& coreClrPath, std::string& hostRuntimeDirectory)
 {
     // If the hosting runtime isn't already set, use the runtime we are debugging
     if (g_hostRuntimeDirectory == nullptr)
     {
-        HRESULT hr = GetCoreClrDirectory(hostRuntimeDirectory);
-        if (FAILED(hr))
+#ifdef FEATURE_PAL
+#if defined(__APPLE__)
+        hostRuntimeDirectory.assign("/usr/local/share/dotnet/shared/Microsoft.NETCore.App");
+#elif defined (__FreeBSD__) || defined(__NetBSD__)
+        ExtErr("FreeBSD or NetBSD not supported\n");
+        return E_FAIL;
+#else
+        // Start with the possible RHEL's locations, then the regular Linux path
+        for (int i = 0; i < _countof(g_linuxPaths); i++)
+        {
+            hostRuntimeDirectory.assign(g_linuxPaths[i]);
+            if (access(hostRuntimeDirectory.c_str(), F_OK) == 0)
+            {
+                break;
+            }
+        }
+#endif
+#else
+        ArrayHolder<CHAR> programFiles = new CHAR[MAX_LONGPATH];
+        if (GetEnvironmentVariableA("PROGRAMFILES", programFiles, MAX_LONGPATH) == 0)
```

Comment:
Is there a reason not to use the desktop runtime on windows? It is guaranteed to be there unlike .Net Core. For someone who acquires SOS automatically via windbg's extension gallery it would be pretty easy for them not to have .Net Core installed. (Not that you need to make a change, but food for thought)

Reply:
I'll keep that in mind, but hosting with the desktop clr is quite a bit different than with coreclr.  The code would be quite a bit different, but you are right that there is always a desktop clr available.